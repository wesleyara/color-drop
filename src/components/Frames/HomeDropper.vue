<script setup lang="ts">
import { useEyeDropper } from "@vueuse/core";

const { isSupported, open, sRGBHex } = useEyeDropper();

const handleDrop = async () => {
  await open();
};
</script>

<template>
  <button :disabled="!isSupported" @click="handleDrop">
    <v-icon class="h-8 w-8" name="bi-eyedropper" />
  </button>

  <span>{{ sRGBHex }}</span>
</template>
