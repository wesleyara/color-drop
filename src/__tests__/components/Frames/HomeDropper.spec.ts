/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, it, expect, beforeEach } from "vitest";
import { mount, VueWrapper } from "@vue/test-utils";
import HomeDropper from "../../../components/Frames/HomeDropper.vue";

describe("HomeDropper unit tests", async () => {
  let sut: VueWrapper<any>;

  beforeEach(async () => {
    sut = mount(HomeDropper);
  });

  it("should render the component", () => {
    const wrapper = sut.exists();

    expect(wrapper).toBeDefined();
  });

  it("should match the snapshot", () => {
    const wrapper = sut.html();

    expect(wrapper).toMatchSnapshot();
  });

  it("should have a icon", () => {
    const wrapper = sut.find("v-icon");

    expect(wrapper.exists()).toBe(true);
  });
});
