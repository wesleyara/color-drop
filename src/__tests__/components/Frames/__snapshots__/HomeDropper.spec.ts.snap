// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeDropper unit tests > should match the snapshot 1`] = `
"<button disabled="">
  <v-icon class="h-8 w-8" name="bi-eyedropper"></v-icon>
</button>
<span></span>"
`;
